mm2src/lp_ordermatch/new_protocol.rs

```diff
@@ -363,4 +382,37 @@ mod new_protocol_tests {
 
         assert_eq!(deserialized, v2);
     }
+
+    #[test]
+    fn check_maker_order_created_serde() {
+        let uuid = Uuid::parse_str("fe5144b8-cb21-4bb8-b235-97211023abd6").unwrap();
+        let compact_uuid = CompactUuid::from(uuid);
+        let conf_settings = OrderConfirmationsSettings {
+            base_confs: 5,
+            base_nota: true,
+            rel_confs: 5,
+            rel_nota: true,
+        };
+        // New format should be serialized to old format when protocol_info is None
+        let maker_order_created_new = MakerOrderCreated {
+            uuid: compact_uuid,
+            base: "BASE".into(),
+            rel: "REL".into(),
+            price: BigRational::from_integer(2.into()),
+            max_volume: BigRational::from_integer(3.into()),
+            min_volume: BigRational::from_integer(1.into()),
+            created_at: 1626639468,
+            conf_settings,
+            timestamp: 1626639468,
+            pair_trie_root: H64::default(),
+            base_protocol_info: None,
+            rel_protocol_info: None,
+        };
+
+        let old_format_str = r#"{"uuid":[254,81,68,184,203,33,75,184,178,53,151,33,16,35,171,214],"base":"BASE","rel":"REL","price":[[1,[2]],[1,[1]]],"max_volume":[[1,[3]],[1,[1]]],"min_volume":[[1,[1]],[1,[1]]],"created_at":1626639468,"conf_settings":{"base_confs":5,"base_nota":true,"rel_confs":5,"rel_nota":true},"timestamp":1626639468,"pair_trie_root":[0,0,0,0,0,0,0,0]}"#;
+
+        let new_format_str = json::to_string(&maker_order_created_new).unwrap();
```

Comment:
P2P messages are serialized to binary format using `rmp_serde` (MessagePack). Please use it instead - testing the `serde_json` doesn't guarantee that MsgPack backward compatibility is not broken.
Please also add the test that the old message format can be deserialized from the new.